### DISCUSSION TITLE
AttributeError during spatial update

### DISCUSSION THREAD

**Discussion by FrancescoPaoloUlloaSeverino** [miniscope/minian#175] [Q&A]
Hi all,

I am getting this error back when running the second step of the first spatial update-parameter exploration of the CNMF section.
I guess some package is missing or not properly installed, but my basic python skills do not allow me to find a solution. Any help?

thanks

AttributeError                            Traceback (most recent call last)
File <timed exec>:21, in <module>

File ~\Anaconda3\envs\minian\lib\site-packages\minian\visualization.py:1898, in visualize_spatial_update(A_dict, C_dict, kdims, norm, datashading)
   1895 hv_A = Dynamic(hv.HoloMap(hv_A_dict, kdims=kdims))
   1896 hv_Ab = Dynamic(hv.HoloMap(hv_Ab_dict, kdims=kdims))
   1897 hv_C = (
-> 1898     hv.HoloMap(hv_C_dict, kdims=kdims)
   1899     .collate()
   1900     .grid("unit_id")
   1901     .add_dimension("time", 0, 0)
   1902 )
   1903 if datashading:
   1904     hv_C = datashade(hv_C)

File ~\Anaconda3\envs\minian\lib\site-packages\holoviews\core\spaces.py:349, in HoloMap.collate(self, merge_type, drop, drop_constant)
    347 from .element import Collator
    348 merge_type=merge_type if merge_type else self.__class__
--> 349 return Collator(self, merge_type=merge_type, drop=drop,
    350                 drop_constant=drop_constant)()

File ~\Anaconda3\envs\minian\lib\site-packages\holoviews\core\element.py:474, in Collator.__call__(self)
    471         self.progress_bar(float(idx+1)/num_elements*100)
    473 components = ndmapping.values()
--> 474 accumulator = ndmapping.last.clone(components[0].data)
    475 for component in components:
    476     accumulator.update(component)

AttributeError: 'NoneType' object has no attribute 'clone'

**Comment by phildong** [comment]
I think this usually happens when all of the cells are dropped during this parameters exploration and the plotting function has no input basically. Could you see if this is the case?
